strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java

```diff
@@ -123,57 +289,97 @@ public void downloadRemotePackageFeed(String storageId,
         Client restClient = proxyRepositoryConnectionPoolConfigurationService.getRestClient();
         try
         {
-            logger.debug(String.format("Downloading remote feed for [%s].", remoteRepositoryUrl));
+            logger.debug(String.format("Downloading NPM changes feed for [%s].", remoteRepositoryUrl));
 
             WebTarget service = restClient.target(remoteRepository.getUrl());
             service = service.path(packageId);
 
-            packageFeed = service.request().buildGet().invoke(PackageFeed.class);
+            InputStream inputStream = service.request().buildGet().invoke(InputStream.class);
+            packageFeed = npmJacksonMapper.readValue(inputStream, PackageFeed.class);
 
-            logger.debug(String.format("Downloaded remote feed for [%s].", remoteRepository.getUrl()));
+            logger.debug(String.format("Downloaded NPM changes feed for [%s].", remoteRepository.getUrl()));
 
         }
         catch (Exception e)
         {
-            logger.error(String.format("Failed to fetch Nuget remote feed [%s]", remoteRepositoryUrl), e);
+            logger.error(String.format("Failed to fetch NPM changes feed [%s]", remoteRepositoryUrl), e);
             return;
         } finally
         {
             restClient.close();
         }
 
         parseFeed(repository, packageFeed);
-
     }
 
     private void parseFeed(Repository repository,
                            PackageFeed packageFeed)
     {
+        try
+        {
+            parseFeedTransactional(repository, packageFeed);
+        }
+        catch (Exception e)
+        {
+            logger.error(String.format("Failed to parse NPM feed [%s/%s]", repository.getRemoteRepository().getUrl(),
```

Comment:
Cosmetic: When the parameters are multi-line, could you please put each of them on a separate line? Thanks! :)